src/client/partition.rs

```diff
@@ -435,6 +486,12 @@ where
                 } => {
                     broker_cache.invalidate().await;
                 }
+                Error::ServerError {
+                    protocol_error: ProtocolError::UnknownTopicOrPartition,
+                    ..
+                } if bind_mode == PartitionClientBindMode::Strong => {
+                    broker_cache.invalidate().await;
+                }
```

Comment:
Why would we not invalidate the cache for the "weak/error" mode of operation?

Reply:
done